code/game/objects/items/devices/translator.dm

```diff
@@ -15,7 +15,10 @@
 /obj/item/device/universal_translator/attack_self(mob/user)
 	if(!listening) //Turning ON
 		langset = input(user,"Translate to which of your languages?","Language Selection") as null|anything in user.languages
-		if(langset)
+		if(langset && ((langset.flags & NONVERBAL) || (langset.flags & HIVEMIND)))
```

Comment:
If someone hits cancel on language selection, does it mean that it will cause it to look like it is on due to adding those checks to the if(langset ...) part?